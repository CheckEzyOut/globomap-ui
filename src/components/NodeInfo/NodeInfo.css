.node-info-box {
  box-sizing: border-box;
  max-width: 50%;
}

.node-info-box {
  display: flex;
  flex-direction: column;
  position: absolute;
  bottom: 30px;
  background-color: white;
  box-shadow: 0 5px 25px rgba(0, 0, 0, 0.2);
  z-index: 99;
}

.node-info-box.left {
  left: 30px;
}

.node-info-box.right {
  right: 30px;
}

.node-info-box .title {
  padding: 30px 25px;
  margin-right: 50px;
}

.node-info-box .title .type {
  display: block;
  font-family: 'opensans-regular', Arial, Helvetica, sans-serif;
  font-size: 1.2em;
  font-weight: bold;
  color: #444;
  margin-bottom: 5px;
}

.node-info-box .title .name {
  font-size: 1.1em;
}

.node-info-box .close-btn {
  position: absolute;
  top: 20px;
  right: 20px;
  border: 0;
  color: #ccc;
  font-size: 1.3em;
  background: none;
}

.node-info-box .tabs {
  display: flex;
  list-style: none;
  padding: 0;
  margin: 0;
}

.node-info-box .tabs li {
  flex: 1;
}

.node-info-box .tabs li button {
  text-transform: uppercase;
  font-family: 'opensans-regular', Arial, Helvetica, sans-serif;
  font-size: 0.8em;
  background: none;
  border: none;
  padding: 15px;
  width: 100%;
  border-bottom: 2px solid #ddd;
}

.node-info-box .tabs li.active button {
  color: #2eb78e;
  border-bottom-color: #2eb78e;
}

.node-info-box .tabs-content {
  display: none;
  height: 250px;
  overflow: auto;
  padding: 20px 0;
}

.node-info-box .tab-content.active {
  display: block;
}
