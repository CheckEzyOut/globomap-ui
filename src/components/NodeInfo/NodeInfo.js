/*
Copyright 2018 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import _ from "lodash";
import React from 'react';
import { connect } from 'react-redux';
import { clearCurrentNode } from '../../redux/modules/nodes';
import { Properties, Monit, Query } from '../';
import './NodeInfo.css';

class NodeInfo extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      tab: 'Properties',
      show: false
    }
  }

  setTab(event, tabName) {
    event.stopPropagation();
    this.setState({ tab: tabName });
  }

  onClose(event) {
    event.stopPropagation();
    return this.props.onClose
            ? this.props.onClose()
            : this.props.clearCurrentNode();
  }

  render() {
    if (!this.props.node)
      return null;

    const node = this.props.node;
    const tabs = [{ name: 'Properties', content: <Properties item={node} /> },
                  { name: 'Monitoring', content: <Monit node={node} /> },
                  { name: 'Reports', content: <Query node={node} /> }];

    const tabsButtons = tabs.map(tabItem => {
      const active = this.state.tab === tabItem.name ? ' active' : '';
      let disabled = false;

      // if (tabItem.name === 'Monitoring' && !['comp_unit', 'zabbix_graph'].includes(node.type)) {
      //   disabled = true;
      // }

      if (tabItem.name === 'Reports') {
        disabled = true;
        let queries = _.filter(this.props.queries, (q) => {
          return q.collection === node.type;
        });
        if (queries.length > 0) {
          disabled = false;
        }
      }

      return (
        <li key={`tab-${tabItem.name}`} className={active}>
          <button className="tab-btn" disabled={disabled}
            onClick={e => this.setTab(e, tabItem.name)}>
            {tabItem.name}
          </button>
        </li>
      );
    });

    const tabsContent = tabs.map(tabItem => {
      const active = this.state.tab === tabItem.name ? ' active' : '';
      return (
        <div key={'content' + tabItem.name} className={'tab-content' + active}>
          {tabItem.content}
        </div>
      );
    });

    let nodeType = node.type;
    if (this.props.namedCollections !== undefined)
      nodeType = this.props.namedCollections[node.type].alias;

    const position = this.props.position ? this.props.position : 'right';
    return (
      <div className={`node-info-box ${position}`}>
        <div className="title">
          <span className="type">{nodeType}</span>
          <span className="name">{node.name}</span>
          <button className="close-btn" onClick={e => this.onClose(e)}>
            <i className="fa fa-times"></i>
          </button>
        </div>
        <ul className="tabs">
          {tabsButtons}
        </ul>
        <div className="tabs-content">
          {tabsContent}
        </div>
      </div>
    );
  }

}

function mapStateToProps(state) {
  return {
    hasId: state.app.hasId,
    queries: state.app.queries,
    namedCollections: state.app.namedCollections
  };
}

export default connect(
  mapStateToProps,
  { clearCurrentNode }
)(NodeInfo);
