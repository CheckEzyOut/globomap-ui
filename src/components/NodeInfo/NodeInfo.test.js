import React from 'react';
import { shallow } from 'enzyme';
import { NodeInfo } from './NodeInfo';

describe("NodeInfo", function() {
  it('should render NodeInfo component', () => {
    const component = shallow(<NodeInfo />);
    expect(component).toMatchSnapshot();
  });
});
