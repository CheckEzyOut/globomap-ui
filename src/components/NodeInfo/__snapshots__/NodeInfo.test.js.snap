// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeInfo should render NodeInfo component 1`] = `""`;
